### ISSUE TITLE
[Bug]: Inconsistent COM Initialization Between MuJoCo's smooth.py and rigid_solver_decomp.py

### ISSUE THREAD

**Issue by pjw971022** [Genesis-Embodied-AI/Genesis#1257]
### Bug Description

There appears to be a discrepancy in how Center of Mass (COM) is initialized between two different implementations in the MuJoCo codebase. This inconsistency may lead to physically incorrect behavior or confusion about the intended COM calculation method.

**Issue Description**
**Location 1: MuJoCo's smooth.py**
In the [smooth.py file](https://github.com/google-deepmind/mujoco/blob/f75772587ac08a71dc4fe3a9cd9fdffd6ec2e2be/mjx/mujoco/mjx/_src/smooth.py#L163), the COM calculation follows the standard MuJoCo approach where individual COM values are computed and stored for each subtree.
**Location 2: rigid_solver_decomp.py _func_COM_links function**
In the _func_COM_links function within rigid_solver_decomp.py, all links' COM are initialized to the same root_COM value:
```python
ti.loop_config(serialize=self._para_level < gs.PARA_LEVEL.ALL)
for i_l_ in range(self.n_awake_links[i_b]):
    i_l = self.awake_links[i_l_, i_b]
    I_l = [i_l, i_b] if ti.static(self._options.batch_links_info) else i_l

    i_r = self.links_info[I_l].root_idx
    self.links_state[i_l, i_b].root_COM = self.links_state[i_r, i_b].root_COM
```

**Potential Issues**

Physical Accuracy: Initializing all links to the same root COM may not reflect the actual physical distribution of mass in the system.
Documentation Inconsistency: The current documentation states:
```
COM=gs.ti_vec3,  # local center of mass of this link
```
However, if link_state.COM is intended to represent the center of mass of the entire entity rather than individual links, this documentation should be updated to reflect the actual behavior.

**Questions for Clarification**

Is the initialization in _func_COM_links intentional, or is this a bug?
If intentional, what is the physical reasoning behind setting all links to the same root COM?
Should the documentation be updated to clarify that COM represents the entire entity's center of mass rather than individual link COM?

### Steps to Reproduce

This can be observed by comparing the COM initialization logic between the two mentioned files and running simulations that depend on accurate COM calculations.

### Expected Behavior

According to MuJoCo's standard implementation in com_pos, each subtree should have its individual COM calculated:
```python
def com_pos(m: Model, d: Data) -> Data:
  """Maps inertias and motion dofs to global frame centered at subtree-CoM."""
  if not isinstance(m._impl, ModelJAX) or not isinstance(d._impl, DataJAX):
    raise ValueError('com_pos requires JAX backend implementation.')
  # calculate center of mass of each subtree
  def subtree_sum(carry, xipos, body_mass):
    pos, mass = xipos * body_mass, body_mass
    if carry is not None:
      subtree_pos, subtree_mass = carry
      pos, mass = pos + subtree_pos, mass + subtree_mass
    return pos, mass
  pos, mass = scan.body_tree(
      m, subtree_sum, 'bb', 'bb', d.xipos, m.body_mass, reverse=True
  )
  cond = jp.tile(mass < mujoco.mjMINVAL, (3, 1)).T
  # take maximum to avoid NaN in gradient of jp.where
  subtree_com = jax.vmap(jp.divide)(pos, jp.maximum(mass, mujoco.mjMINVAL))
  subtree_com = jp.where(cond, d.xipos, subtree_com)
  d = d.replace(subtree_com=subtree_com)
```

### Release version or Commit ID

Mujoco Commit ID: f75772587ac08a71dc4fe3a9cd9fdffd6ec2e2be

### Additional Context

 I may have misunderstood the intended behavior. Please review and provide feedback if this analysis is incorrect. The physical accuracy of COM calculations is crucial for simulation fidelity, so clarification on the intended behavior would be greatly appreciated.

**Comment by duburcqa** [comment]
As you guessed, `link_state.COM` is intended to represent the center of mass of the entire entity rather than individual links. The documentation should indeed be updated accordingly @zswang666 .

**Comment by Kashu7100** [comment]
@duburcqa but if all the `link_state.COM` is intended to represent the COM of the entire entity, `solver.get_links_COM(links_idx)` doesn't make sense... (should be get_entity_COM without links_idx)

**Comment by duburcqa** [comment]
> @duburcqa but if all the link_state.COM is intended to represent the COM of the entire entity, solver.get_links_COM(links_idx) doesn't make sense... (should be get_entity_COM without links_idx)

You are mostly right but not quite. Actually it returns the COM of the root of a link. But an entity could have several roots. For example a single entity may actually comprise several robots connected together with a free joint.

So it should really just be renamed `get_links_root_COM` without any additional change I think.

**Comment by Kashu7100** [comment]
This is more clear.

**Comment by pjw971022** [comment]
I understand. Thank you.

**Comment by pjw971022** [comment]
@duburcqa I have an additional question. I understand that cfrc_ext_ang/cfrc_ext_vel represent external forces acting at the center of mass. If all individual links have identical center of mass positions, would applying an external force (by cfrc_ext_vel) to any link result in the same effect across all links?

**Comment by duburcqa** [comment]
>  If all individual links have identical center of mass positions

They don't. We just do not compute the position of each individual center of mass explicitly.

**Comment by pjw971022** [comment]
Thank you, it was a great help. I will continue with further questions in #1261 . ^^

**Comment by duburcqa** [comment]
> I have an additional question. I understand that cfrc_ext_ang/cfrc_ext_vel represent external forces acting at the center of mass.

cfrc_ext_ang/cfrc_ext_vel represents the external forces acting on each link, expressed at the global center of mass of the whole kinematic tree to which the link belong
